### ISSUE TITLE
Mysql can’t read others value than just float

### ISSUE THREAD

**Issue by DonatoD** [grafana/grafana#36145]
**What happened**:
Using mysql there are problems reading others value than float (like integer, smallinteger, etc)
It is strange because setting as table you can see them. It's like they are treat as string in some way
https://community.grafana.com/t/mysql-cant-read-others-value-than-just-float/48583/4?u=donatod

**What you expected to happen**:
Read these vaue without transform them within the qwery

**How to reproduce it (as minimally and precisely as possible)**:
just create a table with different type of number and show them in a graph

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.0.3
- Data source type & version: phpMyAdmin 5.1.0
- OS Grafana is installed on: Windows 10
- User OS & Browser: Windows 10 & Google

**Comment by zuchka** [comment]
Hi @DonatoD 

What happens if you change your query from ``a `time series` query to a `table` query? I'm asking because there were some breaking changes in v8 that relate to all the SQL data sources. Check out this note and the related issue:

**Comment by DonatoD** [comment]
I'm starting using Graphana deeply by the last version 8, so I cannot say more with previous version (which I used just to learn and make same testing)
Anyway, I made some more test, my table is like this:
```
Timing (timestamp)  |  Item_1 (float) | tem_2  (integer)
2021-06 17:00:00    |  3.58           |  47804
2021-06-17:00:02    |  3.75           |  50908
2021-06-19 17:00:03 |  3.24           |  50918
2021-06-31 17:00:06 |  3.3            |  54065
```

qwery:
```
SELECT
  $__timeGroupAlias(Timing ,$__interval),
  Item_2 
FROM my_table
GROUP BY 1
ORDER BY $__timeGroup(Timing ,$__interval)
```

1) I can see the problem only with non float numbers
2) Selecting integer field columns:
2a) Selecting Time series format, where data are graphed I see a red triangle with "!" inside saying:
_convert value to float failed: metricIndex 1 type can't be converter to float_
But enabling Table view, data are there correctly
2b) Selecting Table format, where data are graphed I read:
_No graphable fields_
But still, enabling Table view, data are there correctly

The only way I found to graph integer data is converting it:
```
SELECT
  $__timeGroupAlias(Timing ,$__interval),
  CONVER(Item_2,DECIMAL) 
FROM my_table
GROUP BY 1
ORDER BY $__timeGroup(Timing ,$__interval)
```
In this way I can see the data as graph both selecting Time series format and Table fomat

**Comment by marefr** [comment]
Duplicate of #35876

**Comment by marefr** [comment]
Fix targeted for v8.0.4 release
